src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -1484,6 +1493,85 @@ bool QgsLayoutItemLegend::isRefreshing() const
 }
 
 
+void QgsLayoutItemLegend::syncLayersWithUpdatedCanvasMinimumMaximum()
+{
+
+  if ( mCustomLayerTree )
```

Comment:
I think we also want to handle this situation - if a user has modified the layers shown they'll still want the legends for those layers to be accurately reflected.